src/libfetchers/git.cc

```diff
@@ -391,6 +400,30 @@ struct GitInputScheme : InputScheme
         AutoDelete delTmpDir(tmpDir, true);
         PathFilter filter = defaultPathFilter;
 
+        RunOptions checkCommitOpts(
+            "git",
+            { "-C", repoDir, "cat-file", "commit", input.getRev()->gitRev() }
+        );
+        checkCommitOpts.searchPath = true;
+        checkCommitOpts.mergeStderrToStdout = true;
+
+        auto result = runProgram(checkCommitOpts);
+        if (WEXITSTATUS(result.first) == 128
+            && result.second.find("bad file") != std::string::npos
+        ) {
+            std::string msg = fmt("Cannot find git-revision in '" ANSI_YELLOW "%s" ANSI_NORMAL
+                    "' of repo '" ANSI_YELLOW "%s" ANSI_NORMAL "' on ref '" ANSI_YELLOW
+                    "%s" ANSI_NORMAL "!\n\n"
+                    "Please make sure that the " ANSI_BOLD "rev" ANSI_NORMAL " exists on the "
+                    ANSI_BOLD "ref" ANSI_NORMAL " you've specified or add " ANSI_BOLD
+                    "allRefs = true;" ANSI_NORMAL " to " ANSI_BOLD "fetchGit" ANSI_NORMAL ".",
```

Comment:
This message seems wrong: `Cannot find git-revision in '<rev>' ...` should be `Cannot find Git revision '<rev>' in ref '<ref>' of repository '<repo>'`.

BTW the ANSI_YELLOW stuff should be unnecessary, it's added automatically if you do `throw Error("... %s ...", bla)`.

Reply:
Fixed, thanks!